packages/file-search/src/browser/quick-editor-open.ts

```diff
@@ -0,0 +1,164 @@
+    protected readonly workspaceService: WorkspaceService;
+
+    @inject(OpenerService)
+    protected readonly openerService: OpenerService;
+
+    @inject(PrefixQuickOpenService)
+    protected readonly prefixQuickOpenService: PrefixQuickOpenService;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    @inject(LabelProvider)
+    protected readonly labelProvider: LabelProvider;
+
+    readonly prefix: string = 'edt ';
+
+    get description(): string {
+        return 'Show All Opened Editors';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            placeholder: '',
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    isEnabled(): boolean {
+        return this.workspaceService.opened;
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get URIs of all EditorWidgets that are currently opened, sorted by URI
+        const widgets = this.applicationShell.widgets
+            .filter((widget): widget is EditorWidget => (widget instanceof EditorWidget))
+            .map(w => w.getResourceUri()!)
+            .sort();
+
+        if (widgets.length === 0) {
+            editorItems.push(new QuickOpenItem({
+                label: 'List of opened editors is currently empty',
+                run: () => false
+            }));
+            acceptor(editorItems);
+            return;
+        }
+
+        for (const uri of widgets) {
+            if (uri.scheme === 'file') {
+                const item = await this.toItem(uri);
+                if (!!token.isCancellationRequested) {
+                    return;
+                }
+                editorItems.push(item);
+            }
+            acceptor(editorItems);
+        }
+        return;
+    }
+
+    private async toItem(uri: URI, group?: QuickOpenGroupItemOptions): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+        let description = this.labelProvider.getLongName(uri.parent);
+        if (this.workspaceService.workspace && !this.workspaceService.workspace.isDirectory) {
+            const rootUri = this.workspaceService.getWorkspaceRootUri(uri);
+            if (rootUri) {
+                description = `${rootUri.displayName} • ${description}`;
```

Comment:
Is it the same way how we show file search results for multi-root workspaces?

Reply:
Yes, this is how file-search implements it. Since the file should be moved to `editor`, which does not have `workspace` as a dependency, I guess it should just display the path without the workspace name (to avoid injecting `workspace`)?

Reply:
yes, we will need to make the label provider multi-root aware if we want to show a root  information in labels application-wide

Reply:
I see. The implementation to show `file-search` results requires injecting `workspace`, which is not a dependency of `editor`. Should we add this dependency, or implement multi-root in an alternative way?

Reply:
Since `editor` pulls `languages`, which pulls `workspace` itself, I added `workspace` in package.json.

Reply:
please revert it, i meant it should be done implicitly via `LabelProvider` custom implementation.  We can improve it later. Editor package used to be independent. We've coupled it already with everything, very soon we can just have one package if we continue to add direct dependencies between everything.

Reply:
I see the idea ;). I just reverted back the part involving `workspace`, so right now it is not multi-root aware. I guess it is good for now and maybe we can improve the `LabelProvider` to support multi-root later?